test/queries/range_query_test.go

```diff
@@ -0,0 +1,160 @@
+// Copyright IBM Corp. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package queries
+
+import (
+	"fmt"
+	"io/ioutil"
+	"math"
+	"testing"
+	"time"
+
+	"github.com/hyperledger-labs/orion-server/internal/worldstate"
+	"github.com/hyperledger-labs/orion-server/pkg/types"
+	"github.com/hyperledger-labs/orion-server/test/setup"
+	"github.com/stretchr/testify/require"
+)
+
+// Scenario:
+// - Create 1 node cluster with QueryLimit = math.MaxInt64
+// - Execute queries with and without pagination triggered by user limit
+// - Execute open intervals queries
+// - Execute queries that returns empty response:
+//// - the start key is after the end key in alphabetical order
+//// - key that does not exist in the database
+func TestRangeQueries(t *testing.T) {
```

Comment:
Add a test in which the pagination is triggered by the server limit. I saw below each record is ~14B, so a limit of say 100B will cause pagination due tot he server limit.

Basically repeat this test, but wit the limit from the server, where the user places no limit.